src/averageAges.js

```diff
@@ -55,7 +65,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(human => onlyWithSon
+    ? people.some(mother => mother.name === human.mother)
+    && human.sex === 'm'
+    : people.find(mother => mother.name === human.mother)
+  );
+
+  const ageDifference = children.map(age =>
+    age.born - people.find(mother => mother.name === age.mother).born
+  );
```

Comment:
Не розривай колбеки, так незрозуміло,
```suggestion
  const ageDifference = children.map(
    age => age.born - people.find(
       mother => mother.name === age.mother
    ).born
  );
```